axon-server-connector/src/main/java/org/axonframework/axonserver/connector/heartbeat/HeartbeatConnectionCheck.java

```diff
@@ -0,0 +1,89 @@
+package org.axonframework.axonserver.connector.heartbeat;
+
+import org.axonframework.axonserver.connector.AxonServerConnectionManager;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import static io.axoniq.axonserver.grpc.control.PlatformOutboundInstruction.RequestCase.HEARTBEAT;
+
+/**
+ * Implementation of {@link ConnectionSanityCheck} which verifies that heartbeat is properly received.
+ *
+ * @author Sara Pellegrini
+ * @since 4.2
+ */
+public class HeartbeatConnectionCheck implements ConnectionSanityCheck {
```

Comment:
As pointed out earlier, I am not entirely sold on using `...Check` as the name. :-(